packages-content-model/roosterjs-content-model-api/lib/modelApi/block/setModelDirection.ts

```diff
@@ -54,12 +65,32 @@ function internalSetDirection(format: ContentModelBlockFormat, direction: 'ltr'
         setProperty(format, 'marginRight', marginLeft);
         setProperty(format, 'paddingLeft', format.paddingRight);
         setProperty(format, 'paddingRight', paddingLeft);
+
+        // If whole Table direction changed, flip cell side borders
+        if (block && block.blockType == 'Table') {
+            block.rows.forEach(row => {
+                row.cells.forEach(cell => {
+                    // Optimise by skipping cells with unchaged borders
+                    updateTableCellMetadata(cell, metadaata => {
+                        if (metadaata?.borderOverride) {
+                            const storeBorderLeft = cell.format.borderLeft;
+                            setProperty(cell.format, 'borderLeft', cell.format.borderRight);
+                            setProperty(cell.format, 'borderRight', storeBorderLeft);
+                        }
+                        return metadaata;
+                    });
+                });
+            });
+
+            // Apply changed borders
+            applyTableFormat(block, undefined, true);
```

Comment:
Please add inline format for the const parameters, for example
applyTableFormat(block, undefined /*....*/, true /*...*/)